Build failure with free-5.0.1

As seen on the Stackage build server:

```
[ 2 of 10] Compiling Text.EDE.Internal.Types ( src/Text/EDE/Internal/Types.hs, dist/build/Text/EDE/Internal/Types.o )

src/Text/EDE/Internal/Types.hs:151:17: error:
    • No instance for (Data.Functor.Classes.Eq1 ExpF)
        arising from the second field of ‘Template’ (type ‘Exp Delta’)
      Possible fix:
        use a standalone 'deriving instance' declaration,
          so you can specify the instance context yourself
    • When deriving the instance for (Eq Template)
    |
151 |     } deriving (Eq)
    |                 ^^
```

Reproduced locally like so:

```bash 
stack unpack ede-0.2.8.7 && cd ede-0.2.8.7
edit stack.yaml # add the following stack.yaml
stack build
```

```yaml
# stack.yaml
resolver: nightly-2018-03-10
extra-deps:
- free-5.0.1
- lens-4.16
```

--- a/src/Text/EDE/Internal/Types.hs
+++ b/src/Text/EDE/Internal/Types.hs
@@ -35,7 +35,6 @@ import           Data.Foldable
 import           Data.HashMap.Strict          (HashMap)
 import           Data.List.NonEmpty           (NonEmpty (..))
 import qualified Data.List.NonEmpty           as NonEmpty
-import qualified Data.List                    as List
 import           Data.Monoid                  (mempty)
 import           Data.Semigroup
 import           Data.Text                    (Text)
@@ -44,6 +43,7 @@ import           Text.PrettyPrint.ANSI.Leijen (Doc, Pretty (..))
 import qualified Text.PrettyPrint.ANSI.Leijen as PP
 import           Text.Trifecta.Delta
 #if MIN_VERSION_base(4,9,0)
+import qualified Data.List                    as List
 import qualified Data.Functor.Classes         as FunctorClasses
 #endif
 

--- a/src/Text/EDE/Internal/Types.hs
+++ b/src/Text/EDE/Internal/Types.hs
@@ -35,6 +35,7 @@ import           Data.Foldable
 import           Data.HashMap.Strict          (HashMap)
 import           Data.List.NonEmpty           (NonEmpty (..))
 import qualified Data.List.NonEmpty           as NonEmpty
+import qualified Data.List                    as List
 import           Data.Monoid                  (mempty)
 import           Data.Semigroup
 import           Data.Text                    (Text)
@@ -42,6 +43,9 @@ import qualified Data.Text                    as Text
 import           Text.PrettyPrint.ANSI.Leijen (Doc, Pretty (..))
 import qualified Text.PrettyPrint.ANSI.Leijen as PP
 import           Text.Trifecta.Delta
+#if MIN_VERSION_base(4,9,0)
+import qualified Data.Functor.Classes         as FunctorClasses
+#endif
 
 -- | Convenience wrapper for Pretty instances.
 newtype PP a = PP { unPP :: a }
@@ -186,6 +190,19 @@ data ExpF a
     | ELoop !Id !a !a
     | EIncl !Text
       deriving (Eq, Show, Functor)
+#if MIN_VERSION_base(4,9,0)
+instance FunctorClasses.Eq1 ExpF where
+    liftEq _ (ELit a) (ELit b) = a == b
+    liftEq _ (EVar a) (EVar b) = a == b
+    liftEq _ (EFun a) (EFun b) = a == b
+    liftEq c (EApp a1 a2) (EApp b1 b2) = a1 `c` b1 && a2 `c` b2
+    liftEq c (ELet a0 a1 a2) (ELet b0 b1 b2) = a0 == b0 && a1 `c` b1 && a2 `c` b2
+    liftEq c (ECase a as) (ECase b bs) = a `c` b && (List.all (uncurry altEq) $ zip as bs)
+        where altEq (pA, a') (pB, b') = pA == pB && a' `c` b'
+    liftEq c (ELoop a0 a1 a2) (ELoop b0 b1 b2) = a0 == b0 && a1 `c` b1 && a2 `c` b2
+    liftEq _ (EIncl a) (EIncl b) = a == b
+    liftEq _ _ _ = False
+#endif
 
 type Exp = Cofree ExpF
 

--- a/stack.yaml
+++ b/stack.yaml
@@ -1,5 +1,5 @@
-compiler:   ghc-8.0.2
-resolver:   nightly-2017-01-25
+compiler:   ghc-8.2.2
+resolver:   lts-11.1
 flags:      {}
 extra-deps: []
 packages:

